src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -148,38 +177,55 @@ private void CheckDisposed()
             }
         }
 
-        private void WriteInternal(byte[] buffer, int offset, int count)
+        private async Task InternalWriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
             if (_chunkedMode)
             {
```

Comment:
It does NOT handle it internally in so far as encoding it.  It simply allows one to omit the 'Content-Length' header and instead use the 'Transfer-Encoding: chunked' header.

I consulted WinHTTP dev lead for verification of my source code for chunking mode.